src/test/kotlin/com/github/codefabrikgmbh/scenarios/GivenTest.kt

```diff
@@ -3,95 +3,95 @@ package com.github.codefabrikgmbh.scenarios
         }.run()
     }
 
-    @Test(expected = SpecialException::class)
-    fun `run without expectation causes exception`() {
-        given(::`test scenario`) {
-            `given the variable is set to`("other")
-            `when this throws an exception`("username.already.taken")
+    @Test(expected = ExpectationException::class)
+    fun `throwing an exception should result in a failed test`() {
+        given(::`variable scenario`) {
+            `when this throws an exception`()
         }.run()
     }
 
     @Test(expected = AssertionError::class)
-    fun `run with expectation without exception`() {
-        given(::`test scenario`) {
-            `given the variable is set to`("other")
-        }.run(SpecialException::class)
+    fun `expecting an exception and none is thrown should result in a failed test`() {
+        given(::`variable scenario`) {
+        }.runExpecting(ExpectationException::class)
     }
 
     @Test
-    fun `run with expectation causes exception`() {
-        given(::`test scenario`) {
-            `given the variable is set to`("other")
+    fun `expecting an exception should work`() {
+        given(::`variable scenario`) {
             `when this throws an exception`("username.already.taken")
-        }.run(SpecialException::class) {
-            `then the user gets the message`(it, "username.already.taken")
+        }.runExpecting(ExpectationException::class) {
+            `then an exception has the message`(it, "username.already.taken")
         }
     }
 
     @Test(expected = AssertionError::class)
-    fun `run with expectation causes wrong exception value`() {
-        given(::`test scenario`) {
-            `given the variable is set to`("other")
+    fun `expecting an exception with a different value should result in a failed test`() {
+        given(::`variable scenario`) {
             `when this throws an exception`("username.already.taken")
-        }.run(SpecialException::class) {
-            `then the user gets the message`(it, "username.not.found")
+        }.runExpecting(ExpectationException::class) {
+            `then an exception has the message`(it, "username.not.found")
         }
     }
 
-    @Test(expected = SpecialException::class)
-    fun `run with expectation causes wrong exception type`() {
-        given(::`test scenario`) {
-            `given the variable is set to`("other")
+    @Test(expected = ExpectationException::class)
+    fun `expecting an exception with a different type should result in a failed test`() {
+        given(::`variable scenario`) {
             `when this throws an exception`("username.already.taken")
-        }.run(RuntimeException::class)
+        }.runExpecting(RuntimeException::class)
     }
 
     @Test
-    fun `run with expectation succeeding run check`() {
-        given(::`test scenario`) {
-            `given the variable is set to`("other")
+    fun `testing something after expecting an exception should work`() {
```

Comment:
Please define 'something' more explicitly like `asserting a valid value after expecting an exception should work` and `asserting an invalid value after expecting an exception should result in a failed test`.  Without defining 'something' i can not evaluate if the test is correctly implement - testing explicitly what has been intended to test.